File: PYTHON/SwapNumbers.py
a = int(input("Enter first number: "))
b = int(input("Enter second number: "))

print("\n With 3rd variable")
print(f"Before: a={a} b={b}")

temp = a
a = b
b = temp
print(f"After: a={a} b={b}")

print("\n Without 3rd variable")
print(f"Before: a={a} b={b}")
a = a + b
b = a - b
a = a - b
print(f"After: a={a} b={b}")

File: C/SwapNumbers.c
#include <stdio.h>

int main() {
    int a, b, temp;
    
    printf("Enter first number: ");
    scanf("%d", &a);
    printf("Enter second number: ");
    scanf("%d", &b);
    
    printf("\n With 3rd variable\n");
    printf("Before: a=%d b=%d\n", a, b);
    
    temp = a;
    a = b;
    b = temp;
    printf("After: a=%d b=%d\n", a, b);
    
    printf("\n Without 3rd variable\n");
    printf("Before: a=%d b=%d\n", a, b);
    a = a + b;
    b = a - b;
    a = a - b;
    printf("After: a=%d b=%d\n", a, b);
    
    return 0;
}
